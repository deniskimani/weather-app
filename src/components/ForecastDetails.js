import "../styles/ForecastDetails.css";

function ForecastDetails(props) {
  const { forecast } = props;
  const formattedDate = new Date(forecast.date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>

      <div className="forecast-details__max--temp">
        <p>
          <span>MaxTemperature:</span> {forecast.temperature.max} &deg;C
        </p>
      </div>
      <div className="forecast-details__min--temp">
        <p>
          <span>Min Temperature:</span> {forecast.temperature.min} &deg;C
        </p>
      </div>

      <div className="forecast-details__humidity">
        <p>
          <span>Humidity:</span> {forecast.humidity} &#37;
        </p>
      </div>

      <div className="forecast-details__wind">
        <p>
          <span>Wind:</span> {forecast.wind.speed} mph
        </p>
      </div>
    </div>
  );
}
export default ForecastDetails;
