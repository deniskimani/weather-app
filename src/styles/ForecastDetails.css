.forecast-details {
    text-align: center;
    padding: 20px;
  }
.forecast-details .forecast-details__date{
    font-size: 24px;
    font-weight: 700;

}

span{
    font-size: 17px;
    font-weight: 500px !important;
}