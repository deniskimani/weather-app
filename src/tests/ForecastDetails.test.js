import { render } from "@testing-library/react";

import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description 1",
    icon: "stubIcon1",
    temperature: {
      max: 22,
      min: 12,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
    humidity: 30,
  };
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
