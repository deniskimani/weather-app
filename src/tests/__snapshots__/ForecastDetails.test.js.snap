// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__max--temp"
    >
      <p>
        <span>
          MaxTemperature:
        </span>
         22 °C
      </p>
    </div>
    <div
      class="forecast-details__min--temp"
    >
      <p>
        <span>
          Min Temperature:
        </span>
         12 °C
      </p>
    </div>
    <div
      class="forecast-details__humidity"
    >
      <p>
        <span>
          Humidity:
        </span>
         30 %
      </p>
    </div>
    <div
      class="forecast-details__wind"
    >
      <p>
        <span>
          Wind:
        </span>
         13 mph
      </p>
    </div>
  </div>
</DocumentFragment>
`;
